Source/WebCore/style/values/anchor-position/StylePositionTryFallback.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "StylePositionTryFallback.h"
+
+#include "StyleBuilderChecking.h"
+#include "StylePrimitiveKeyword+CSSValueConversion.h"
+#include "StylePrimitiveKeyword+CSSValueCreation.h"
+#include "StylePrimitiveKeyword+Logging.h"
+#include "StylePrimitiveKeyword+Serialization.h"
+#include "StylePropertiesInlines.h"
+
+namespace WebCore {
+namespace Style {
+
+bool PositionTryFallback::PositionArea::operator==(const PositionTryFallback::PositionArea& other) const
+{
+    if (properties && other.properties) {
+        if (properties == other.properties)
+            return true;
+
+        Ref strongProperties = *properties;
+        Ref strongOtherProperties = *other.properties;
+
+        auto lhsPositionArea = strongProperties->getPropertyCSSValue(CSSPropertyPositionArea);
+        ASSERT(lhsPositionArea);
+
+        auto rhsPositionArea = strongOtherProperties->getPropertyCSSValue(CSSPropertyPositionArea);
+        ASSERT(rhsPositionArea);
+
+        return *lhsPositionArea == *rhsPositionArea;
+    }
+
+    return !properties && !other.properties;
+}
+
+bool PositionTryFallback::operator==(const PositionTryFallback& other) const
+{
+    if (positionArea.properties && other.positionArea.properties)
+        return positionArea == other.positionArea;
+
+    if (!positionArea.properties && !positionArea.properties)
```

Comment:
Consider why our tests didn't catch this mistake?
```suggestion
    if (!positionArea.properties && !other.positionArea.properties)
```